.github/workflows/test.yml

```diff
@@ -13,7 +13,57 @@ jobs:
       - uses: denoland/setup-deno@v1
         with:
           deno-version: v1.x
-      - run: KAKAROT_ADDRESS=0x1 deno test --allow-env
+      # Checkout target branch and run tests
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+      - name: Run Tests with Coverage on target branch
+        run: KAKAROT_ADDRESS=0x1 deno test --allow-env --coverage=cov_profile
+      - name: Generate Coverage Report on target branch
+        run: deno coverage cov_profile --html
+      - name: Install bc
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y bc
+      - name: Extract coverage percentage from HTML for the target branch
+        run: |
+          cat cov_profile/html/index.html
+          coverage_percentage=$(grep -A 3 "<div class='fl pad1y space-right2'>" cov_profile/html/index.html | grep "<span class='strong'>" | awk -F'[<>]' '{gsub("%","",$3); print $3}')
+          echo $coverage_percentage > curr_coverage_percentage.txt
+      - name: Upload current coverage percentage
+        uses: actions/upload-artifact@v4
+        with:
+          name: curr-coverage-percentage
+          path: curr_coverage_percentage.txt
```

Comment:
Fixed, I've used a different approach that I didn't see earlier. 
I used artifacts because the only two workarounds I had found to prevent actions/checkout from deleting all previously created files were: artifacts or the previous method with git pull ecc. but I replaced it with clean:false. Thanks